service/controller/clusterapi/v29/resource/clusterazs/resource.go

```diff
@@ -29,10 +32,14 @@ func New(config Config) (*Resource, error) {
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
+	if config.ToClusterFunc == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ToClusterFunc must not be empty", config)
+	}
 
 	r := &Resource{
-		cmaClient: config.CMAClient,
-		logger:    config.Logger,
+		cmaClient:     config.CMAClient,
+		logger:        config.Logger,
+		toClusterFunc: config.ToClusterFunc,
```

Comment:
Is it used or this is some prep?

Reply:
Both, we use it during testing and prepare the machine deployment controller for the node pool reconciliation. Not sure if this answers your question though.